### PR TITLE
[Multitenancy] functionality of webhooks, labels and image_url per wallet

### PR DESCRIPTION
**Pull request by baegjae**
### Description:
The webhook functionality is an essential feature, because the controller should receive messages from the aca-py server. This PR enables the functionality of webhooks, labels and image_url per wallet. Also, it contains useful other features. See updates and APIs changes below.

### Terms:
**Base wallet**: Wallet created when aca-py is launched.

### Updates:
1. `webhook_urls` works per wallet
2. `label` works per wallet
3. `image_url` works per wallet
4. on demand wallet loading (reduce boot-up time)
5. wallet mapping (`connection_id` or `key` -> `wallet_name`) is stored in the storage (preparation for scale-out aca-py)
6. When registering nym, optional `wallet_name` is used to register the did of another wallet
7. When creating local did, optional `seed` is used

### APIs changes:
- `POST /wallet` : Add new wallet (added `webhook_urls`, `label`, `image_url`) - public
- `GET /wallet` : Get information of wallets with query - base wallet only
- `PUT /wallet/{wallet_id}` : Update information of a wallet - base wallet only
- `DELETE /wallet/{wallet_id}` : Remove a wallet - base wallet only
- `GET /wallet/me` : Get information of my wallet - user wallet
- `PUT /wallet/me` : Update information of my wallet - user wallet
- `DELETE /wallet/me` : Remove my wallet - user wallet
- `/wallet/did/create` : Create a local DID (added `seed`)
- `/ledger/register-nym` : Send a NYM registration to the ledger (added `wallet_name` to register the did of another wallet)

### Demo changes:
- The `--multitenant` flag enables multitenancy support. With this, Faber and Alice will use a single Base agent. Please see the detailed flows in README-multitenant.md.

### TODO
- Test code updates (Several `send_webhook` functions included `context` as a parameter to make webhook works per wallet)

### CODE REVIEW

**Comment by ianco** [comment]
Can you provide some info on how to test this code?

It's tough to asses without any unit tests or a working demo ...

Thx

**Comment by baegjae** [comment]
@ianco I pushed the additional commits to updates faber/alice demo. Also, I updated the description for the demo changes. Thanks!

File: demo/README-multitenant.md [hunk 1]
```diff
@@ -6,25 +6,24 @@ To run either Alice or Faber with multitenancy enabled run with a `--multitenant
...
+and:
 
 ```bash
 ./run_demo alice --multitenant
 ```
 
-This will enable a couple of additional options within the demo:
+This will enable a multitenancy support, and Faber and Alice will use a single Base agent.
 
-* `W` to create a new wallet (or switch to an existing wallet)
```
Comment (ianco) [inline demo/README-multitenant.md:17]: Not sure why you removed these options?  I added them so you could test multi-wallet scenarios, e.g. faber issuing from one wallet and then verifying from another, alice receiving credentials in different wallets, etc.

File: demo/runners/alice.py [hunk 2]
```diff
@@ -229,32 +230,41 @@ async def main(
...
 
         log_status("#9 Input faber.py invitation details")
         await input_invitation(agent)
 
         options = (
             "    (3) Send Message\n"
             "    (4) Input New Invitation\n"
-        )
-        if multitenant:
-            options += "    (W) Create and/or Enable Wallet\n"
```
Comment (ianco) [inline demo/runners/alice.py:255]: See comment from above, I'd prefer to keep these options in the demo

File: demo/runners/faber.py [hunk 3]
```diff
@@ -254,16 +311,6 @@ async def main(
             if option is None or option in "xX":
                 break
 
-            elif option in "wW" and multitenant:
```
Comment (ianco) [inline demo/runners/faber.py:257]: Same comment here.

**Review by ianco — CHANGES_REQUESTED** [review]

File: demo/README-multitenant.md (hunk 1 above)
Comment (baegjae) [reply demo/README-multitenant.md]: Oh, I'm sorry. I didn't think much about it. I restored these features.

**Comment by baegjae** [comment]
I restored additional options for creating a new wallet in faber/alice demo. I agree that these options are effective in demonstrating the features of multitenancy.

File: demo/runners/alice.py (hunk 2 above)
Comment (baegjae) [reply demo/runners/alice.py]: Restored the options.

File: demo/runners/faber.py (hunk 3 above)
Comment (baegjae) [reply demo/runners/faber.py:257]: Restored the options. Thanks.

**Review by ianco — APPROVED** [review]

**Pull request merged**